src/core/toga/window.py

```diff
@@ -34,6 +34,11 @@ def __init__(self, id=None, title=None,
         self._size = size
         self._is_full_screen = False
 
+        self._closed = False
```

Comment:
So, a window won't be visible unless `show()` is actually called (which will definitely be the *common* use case, but isn't strictly *required*). However, the state logic around `self._closed`:
* Assumes the window will be opened
* Only allows you to show a window if it is already not-closed
* Will allow you to close a window before it's been shown.

Reply:
Yes. I think I heard somewhere (maybe on the gitter channel) that on some platforms you can't re-open a closed window. So I was trying to express that in the code somehow. But I'm open to suggestions on what else we should do.

I've thought of some options:

1. Just don't have self._closed. If the user tries to re-open a closed window, they get to keep both pieces.
2. Document in a docstring for show() that trying to show a window that has been closed is possibly not supported on all platforms
3. Make self._closed a ternary variable. It defaults to None. During on_close() self._closed is set to True. During show() if it's True, throw an exception, or silently exit, otherwise set it to False, and continue.

But to be honest, option 3 is sounding complicated, and I'm not sure it's correct. Also, it's entirely possible for a programmer to replace app.on_close(), so maybe we can't depend on that?

Reply:
Closing a window.forms.Window under MS Windows is a destructive act; however, that's something that affects the *native* layer, not the public interface API. The public interface is (or should be) consistent, and the "conceptual" window can be re-opened and have preserved state.

I can see how a state variable could be useful - "is this window visible?" is definitely a question that might need to be answered, and possibly surfaced as an end-user API.

My concern here is that the `_closed` variable doesn't actually reflect the state of anything - the way the logic is currently being handled, the value of `_closed` is the opposite of the actual state of the window at several points (e.g., it's created in an "open" state, even though the window isn't actually open yet). What I'm advocating for is a state variable that actually reflects the state of the window.

Reply:
I'm happy to remove the _closed variable. Your reply about MS Windows native layer closing being a destructive act re-assures me that having a closed() function doesn't need a _closed variable.

Perhaps if I do another commit that removes the _closed variable, this PR might be ready for inclusion, and we can start a new PR to discuss adding a window.state variable?